File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    # n : 노드 수
    # computers : 연결 정보 2차원 배열
    answer = 0
    graph = [[] for _ in range(n+1)]
    for i in range(len(computers)):
        for j in range(len(computers[i])): 
            if computers[i][j]==1: 
                graph[i+1].append(j+1)
                
    def DFS(node):
        # 방문처리
        visited[node] = True
        for x in graph[node]:
            # 아직 방문 안했으면 고
            if visited[x]==False:
                DFS(x)
    
    visited = [False for _ in range(n+1)]
    
    for i in range(1, n+1):
        if visited[i]==False:
            DFS(i)
            answer +=1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    // 네트워크 수 세기
    let networks = 0;

    // 방문 여부 확인하기
    let visit = Array(n).fill(false);
    
    // 방문한 곳인지 확인 및 네트워크 수 증가시키기
    for (let i = 0; i < n; i++) {
        if (!visit[i]) {
            networks ++;
            isVisit(i);
        }
    }
    
    // DFS를 통해 방문한 곳인지 확인하는 함수
    function isVisit(index) {
        visit[index] = true;
        
        const computer = computers[index];
        
        for (let i = 0; i < computer.length; i++) {
            const isConnect = computer[i] === 1 ? true : false;
            
            // 방문하지 않았고 && 연결된 곳인지 확인
            if (!visit[i] && isConnect) {
                isVisit(i);
            }
        }        
    }
    
    return networks;
}